File: core/ByoSnapCSharp/Controllers/HealthController.cs
using Microsoft.AspNetCore.Mvc;

namespace ByoSnapCSharp.Controllers
{
  [ApiController]
  [ApiExplorerSettings(IgnoreApi = true)]
  public class HealthController : ControllerBase
  {
    // @GOTCHAS - Health Check Endpoint
    //   1. The health URL does not take any URL prefix like other APIs
    [HttpGet("healthz")]
    public IActionResult HealthCheck()
    {
      return Ok("Ok");
    }
  }
}

File: core/byosnap-java/src/main/java/com/snapser/byosnap/controller/HealthController.java
package com.snapser.byosnap.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * Readiness / health check.
 *
 * <p>@GOTCHAS - Health Check Endpoint
 * <ol>
 *   <li>The health URL does NOT take any URL prefix like the other APIs. It is
 *       served at the root ({@code /healthz}).</li>
 * </ol>
 */
@RestController
public class HealthController {

    @GetMapping("/healthz")
    public ResponseEntity<String> health() {
        return ResponseEntity.ok("Ok");
    }
}
